Developer-Guide--Writing-Tests.md

```diff
@@ -311,3 +317,142 @@ end
 In Markus, one use case of shared examples is [`a criterion`](https://github.com/MarkUsProject/Markus/blob/master/spec/support/criterion.rb), which specifies the common behavior of `RubricCriterion` and a `FlexibleCriterion`.
 
 You can also use an alias for the method `it_behaves_like_a` to make the spec code read better. For example, `it_has_behavior 'enumerability'`. The aliases should be defined in [spec/support/it_behaves_like_aliases.rb](https://github.com/MarkUsProject/Markus/blob/master/spec/support/it_behaves_like_aliases.rb).
+
+## Testing with Jest
+
+[Jest](https://jestjs.io/) is a JavaScript testing framework focused on simplicity. It works well with multiple popular frameworks, such as Node, Angular, Vue, and of course React. It provides a test runner and several handy functions; however, to fully use its capability, Jest is often combined with some other testing libraries. As of now, both [React Testing Library (RTL)](https://testing-library.com/docs/react-testing-library/intro/) and [Enzyme](https://enzymejs.github.io/enzyme/) are being used alongside Jest in Markus.
+
+On a high level, RTL allows you to test from the user's perspective, while Enzyme gives you access to the internal states/implementation of the components.
+
+Similar to RSpec, Jest also involves specifications. It is recommended that you follow along the `__tests__` folder, under `app/assets/javascripts/Components/`, for this tutorial.
+
+## How to Run Jest Specifications
```

Comment:
Every heading from this one to the end should be increased one level (e.g., `###` instead of `##` here) because they're nested under "Testing with Jest".